soc/arm/atmel_sam0/common/soc_power.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2019 Benjamin Valentin
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <soc_power.h>
+#include <soc.h>
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+
+static inline void _idle(uint8_t mode) {
```

Comment:
I'm a bit confused myself about usage rules of leading underscore in function names, however we generally don't use them for static functions.

Reply:
I renamed the function to `sam0_idle()`